app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt

```diff
@@ -49,35 +60,64 @@ class PoliciesFragmentPresenter @Inject constructor(
+      ).parseOppiaHtml(
+        policyDescription,
+        binding.policyDescriptionTextView,
+        supportsLinks = true,
+        supportsConceptCards = false
+      )
+
+      binding.policyWebLinkTextView.text = htmlParserFactory.create(
+        gcsResourceName = "",
+        entityType = "",
+        entityId = "",
+        imageCenterAlign = false,
+        customOppiaTagActionListener = null,
+        resourceHandler.getDisplayLocale()
+      ).parseOppiaHtml(
+        policyWebLink,
+        binding.policyWebLinkTextView,
+        supportsLinks = true,
+        supportsConceptCards = false
+      )
     } else if (policyPage == PolicyPage.TERMS_OF_SERVICE) {
       policyDescription =
         resourceHandler.getStringInLocale(R.string.terms_of_service_content)
-      policyWebLink = resourceHandler.getStringInLocale(R.string.terms_of_service_web_link)
-    }
-
-    binding.policyDescriptionTextView.text = htmlParserFactory.create(
-      policyOppiaTagActionListener = this,
-      displayLocale = resourceHandler.getDisplayLocale()
-    ).parseOppiaHtml(
-      policyDescription,
-      binding.policyDescriptionTextView,
-      supportsLinks = true,
-      supportsConceptCards = false
-    )
+      val spannableString = SpannableString(HtmlCompat.fromHtml(policyDescription, HtmlCompat.FROM_HTML_MODE_LEGACY))
+      val uri = "https://www.oppia.org"
+      val spanStart = policyDescription.indexOf(uri) - 4
+      val spanEnd = spanStart + uri.length
+      val clickableSpan = object : ClickableSpan() {
+        override fun onClick(widget: View) {
+          val intent = Intent(Intent.ACTION_VIEW, Uri.parse(uri))
+          activity.startActivity(intent)
+        }
 
-    binding.policyWebLinkTextView.text = htmlParserFactory.create(
-      gcsResourceName = "",
-      entityType = "",
-      entityId = "",
-      imageCenterAlign = false,
-      customOppiaTagActionListener = null,
-      resourceHandler.getDisplayLocale()
-    ).parseOppiaHtml(
-      policyWebLink,
-      binding.policyWebLinkTextView,
-      supportsLinks = true,
-      supportsConceptCards = false
-    )
+        override fun updateDrawState(ds: TextPaint) {
+          super.updateDrawState(ds)
+          ds.isUnderlineText = true
+        }
+      }
+      spannableString.setSpan(clickableSpan, spanStart, spanEnd, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+      binding.policyDescriptionTextView.text = spannableString
+      binding.policyDescriptionTextView.movementMethod = LinkMovementMethod.getInstance()
+      binding.policyWebLinkTextView.text = htmlParserFactory.create(
+        gcsResourceName = "",
+        entityType = "",
+        entityId = "",
+        imageCenterAlign = false,
+        customOppiaTagActionListener = null,
+        resourceHandler.getDisplayLocale()
+      ).parseOppiaHtml(
+        policyWebLink,
+        binding.policyWebLinkTextView,
+        supportsLinks = true,
+        supportsConceptCards = false
+      )
+    }
```

Comment:
DRY: This code is identical to the if branch above. Can you reduce duplication by moving it out of the branches, or to a function?

Reply:
Sure